filesystem/resource_filesystem_directory_test.go

```diff
@@ -0,0 +1,130 @@
+package filesystem
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"regexp"
+	"syscall"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccFilesystemDirectory(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers: map[string]terraform.ResourceProvider{"filesystem": Provider()},
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				PreConfig: func() { os.Remove("/tmp/test") },
+				Config: `
+resource "filesystem_directory" "test" {
```

Comment:
For better readability, I would probably move HCL configuration strings to constants